workflows/Snakefile

```diff
@@ -33,56 +34,60 @@ rule all:
 
 
 
-rule parametrize_mock_data:
-    """Define parameterization of mock data generation."""
+rule make_random_or_deep_tree:
+    """Make a tree (TreeNode) of random/deep topology and save it as JSON."""
     input:
-        expand('../data/mock/seed_{seed}_n_trees_{n_trees}_n_cells_{n_cells}_n_mutations_{n_mutations}_fpr_{fpr}_fnr_{fnr}_na_rate_{na_rate}_observe_homozygous_{observe_homozygous}_strategy_{strategy}_tree_{tree_no}.json',
-            seed=params_md["seed"],
-            n_trees=params_md["n_tree"],
-            n_cells=params_md["n_cells"],
-            n_mutations=params_md["n_mutations"],
-            strategy=params_md["strategy"],
-            fpr=params_md["fpr"],
-            fnr=params_md["fnr"],
-            na_rate=params_md["na_rate"],
-            observe_homozygous=params_md["observe_homozygous"],
-            tree_no= list(range(1, params_md["n_tree"]+1))
-        )
-
-
-rule generate_mock_data:
-    """Generates mock data using wildcards defined in rule:parametrize_mock_data"""
+        script="../scripts/make_tree.py"
+    output:
+        tree="../data/{experiment}/trees/T_{tree_type}_{n_nodes}_{tree_seed}.json"
+    shell:
+        """
+        poetry run python {input.script} \
```

Comment:
To make it work with miniconda, you can remove `poetry run` from the Snakefile and work *within* `poetry shell`, which is similar to miniconda's environment.

Reply:
I tried just now -  but didn't get it to work. My console I run snakemake from gets caught in the poetry shell.